docs/wiki/GCR-Benchmarking.md

```diff
@@ -0,0 +1,80 @@
+```
+DOCKER_HUB_USERNAME=<your_docker_hub_username>
+DOCKER_HUB_ACCESS_TOKEN=<your_docker_hub_access_token>
+```
+
+In addition, STeLLAR requires two core components to deploy and benchmark serverless functions: The function code and a JSON file specifying experiment parameters.
+
+### Function code
+
+Put your code directory at `src/setup/deployment/raw-code/serverless/gcr/<function_code_dir>`. Ensure that the Dockerfile is at the root of the `<function_code_dir>` directory. This function code must comply with [Google Cloud's Container runtime contract](https://cloud.google.com/run/docs/container-contract).
+
+As Cloud Run is [based on Knative](https://cloud.google.com/blog/products/serverless/knative-based-cloud-run-services-are-ga), any Knative applications can be run out of the box on Cloud Run. See [here](https://knative.dev/docs/samples/serving/) for examples of code for various runtimes.
+
+### Experiment JSON file
+
+The JSON file specifying configurations can be placed at any path. The folder for our examples can be located under the `experiments/` folder. For examples on how to write the experiment JSON file, see [here](https://github.com/vhive-serverless/STeLLAR/tree/feature-serverless-framework-deployment/experiments/tests/gcr).
+
+Summary of experiment JSON file values:
+
+| Key | Type | Description |
+| --- | --- | --- |
+| Sequential | boolean | Specifies whether to run sub-experiments sequentially or in parallel |
+| Provider | string | Specifies the Cloud Provider to deploy the function to. (Possible values: `aws`, `azure`, `gcr`, `cloudflare` |
+| Runtime | string | Specifies the runtime of the function. (e.g. `go1.x`, `java11`, etc.) |
+| SubExperiments | array | Array of objects specifying specific experiment parameters for each sub-experiment |
+| Title | string | Name of the sub-experiment |
+| Function | string | Name of the function. This must be the same value as `<function_code_dir>` |
+| Handler | string | Entry point of the function. For GCR, the value must be `Dockerfile` |
+| PackageType | string | Specifies the type of packaging for function upload. For GCR, the value must be `Container` |
+| Bursts | number | Specifies the number of bursts to send to the deployed function(s) |
+| BurstSizes | array | Specifies the size of each burst when invoking the deployed function(s). STeLLAR iterates and cycles through the array for each burst. |
+| IATSeconds | number | Specifies the interarrival time between each burst. |
+| DesiredServiceTimes | array | Specifies the desired service execution time(s) when invoking the deployed function(s). STeLLAR iterates and cycles through the array for each burst. These execution times are achieved by calculating the corresponding busy spin count for the desired time on the **host running the STeLLAR client**. |
+| FunctionImageSizeMB | number | Specifies the target size of the function to upload. Currently only supported for ZIP file deployment under AWS. |
```

Comment:
```suggestion
| FunctionImageSizeMB | number | Specifies the target size of the function to upload. |
```
I think we can exclude the part about only having support for AWS, since we also run image size experiments on GCR now. Perhaps in the future when we write the Wiki page for Cloudflare we can consider just removing the parameter entirely from that Wiki.

Reply:
Good catch, thanks!